Added checks and try to use key B for hf mf dump cmd

chuayupeng opened the pull request:
Was trying to clone a card that only allowed key B read and write access, had to do changes to the code to make it work, thought I should do a PR for that. Relevant issue here: https://github.com/Proxmark/proxmark3/issues/738

pwpiwi commented on client/cmdhfmf.c:
```diff
@@ -331,11 +331,17 @@ int CmdHF14AMfDump(const char *Cmd)
-			memcpy(c.d.asBytes, keyA[sectorNo], 6);
-			SendCommand(&c);
-
+		bool doesKeyAWork = true;
```
This is not necessary, The sector trailer can either be read by Key A or not at all. There is no access condition which allows reading by Key B only.

pwpiwi commented on client/cmdhfmf.c:
```diff
@@ -365,13 +373,21 @@ int CmdHF14AMfDump(const char *Cmd)
-					memcpy(c.d.asBytes, keyA[sectorNo], 6);
-					SendCommand(&c);
-					received = WaitForResponseTimeout(CMD_ACK,&resp,1500);
+					if(doesKeyAWork){
```
See comment above: At least the Access Conditions can always be read with key A.

pwpiwi commented on client/cmdhfmf.c:
```diff
@@ -365,13 +373,21 @@ int CmdHF14AMfDump(const char *Cmd)
+					}
 				} else {												// data block. Check if it can be read with key A or key B
 					uint8_t data_area = sectorNo<32?blockNo:blockNo/5;
 					if ((rights[sectorNo][data_area] == 0x03) || (rights[sectorNo][data_area] == 0x05)) {	// only key B would work
```
Key B is used below.

pwpiwi commented:
The existing code already uses Key B where necessary. See review comments. Did I miss anything?

chuayupeng commented:
Not entirely sure why but the existing code didn't work for the card I was trying to dump even though I could rdbl with key B, so I added the changes to make it work for my case. If it appears unnecessary or superfluous it's only because in my use case the conditions to use key B were never triggered, so I added in the clauses while keeping the overall structure of the function the same.
